src/components/edit/Agents.jsx

```diff
@@ -59,7 +61,8 @@ const EditAgents = (props) => {
       .then(agent => setSelectedAgent(agent));
   }
 
-  const onSelectChange = (selected) => {
+  const onSelectChange = (selected, field) => {
+    selected.field = field;
```

Comment:
it's not good practice to mutate inputs -- you don't need to attach this to `field` -- instead, on the next line, you can do `[field]: selected.value`